### ISSUE TITLE
$'\r' command not found

### ISSUE THREAD

**Issue by antho214** [HenriquesLab/DL4MicEverywhere#45]
**Describe the bug**

Launching the `Windows_launch.bat` immediately closes the terminal. Adding a pause at the end of the `Windows_launch.bat` file allows to see the error (see screenshot section). The error seems to be related to the return escape character (`\r`). The bug appears on two Windows machines in our lab.

**To Reproduce**

1. Go to 'DL4MicEverywhere' folder
2. Double-Click on 'Windows_launch.bat'
3. Terminal opens; prints the error; then closes
4. See screenshot obtained after adding a pause at the end of the `Windows_launch.bat` file.

**Expected behavior**

I would expect the file to launch the Linux environment and see the DL4MicEverywhere windows.

**Screenshots**

![Screenshot 2024-05-27 144842](https://github.com/HenriquesLab/DL4MicEverywhere/assets/34163910/6651de32-ab21-4c81-9f59-09cc313cf48f)

**Desktop (please complete the following information):**

 - OS: Windows
 - Browser: N/A
 - Version: 

```bash
> wsl -v
WSL version: 2.2.4.0
Kernel version: 5.15.153.1-2
WSLg version: 1.0.61
MSRDC version: 1.2.5326
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.26091.1-240325-1447.ge-release
Windows version: 10.0.22631.3593
```

**Additional context**

I was able to solve this issue on both computers by manually installing and converting every files within the repository using `dos2unix` (https://linux.die.net/man/1/dos2unix). 

The procedure was 
```bash
> wsl -d Ubuntu
> sudo apt update
> sudo apt install dos2unix
> find . -type f -print0 | xargs -0 dos2unix
```

**Comment by IvanHCenalmor** [comment]
Hi @antho214, 

First of all, thanks a lot for all the feedback and making it so well detailed 😄 

I think that I might now what has happened. Git has an automatic conversion of end of lines. The script is expecting for '\n' end of lines, but with the auto-conversion those end of lines are converted into '\r' and I think that this thing is creating the issue. There are ways to deactivate this auto-conversion, but I understand that maybe people want to keep that option for other projects.

So yeah I think that your proposed solution looks really nice, installing dos2unix could be the best solution to avoid this type of problems. 

Thank you very much for your input, I will update the code to fix this bug! 🤗

**Comment by IvanHCenalmor** [comment]
Hi @antho214,

I finally found time to work in the solution for this issue. The commit [fb7ce36](https://github.com/HenriquesLab/DL4MicEverywhere/commit/fb7ce364e4cf88cc2664da6203b1e9c51f6b9dab) should fix it.

At the end I followed your suggestion:
 - First, we check if dos2unix is already installed and if not we install it
 - Then, it checks if one of the files is in the DOS/Windows format and if so it convert all of them using dos2unix

Thank you very much for your help and reporting this issue! 🤗
